lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -184,7 +190,26 @@ export default function LMSFilePicker({
         if (pathChanged) {
           return;
         }
-        setDialogState({ state: 'fetched', files });
+        if (apiSendsChildren) {
+          // If the backend is sending all the tree structure
+          // append these to the top level `__root___` folder.
+          folderPath[0].children = files;
+        }
+
+        if (apiSendsChildren && isReload) {
+          let reloaded_files = files;
+          // Iterate over the paths skipping the first `__root__` elmement
+          for (let i = 1; i < folderPath.length; i++) {
+            const path = folderPath[i];
+            // Find this level's path in the files returned
+            const foundPath = reloaded_files.find(file => file.id === path.id);
+            reloaded_files = foundPath ? foundPath.children : [];
+          }
+          // Set the files at the correct path
+          setDialogState({ state: 'fetched', files: reloaded_files });
```

Comment:
We don't handle the case when the location you are in dispears when you reload.

We don't handle it either for the one-fetch-per-folder case I reckon so we could tackle that separately.